ledger-service/http-json/src/main/scala/com/digitalasset/http/EndpointsCompanion.scala

```diff
@@ -245,17 +259,31 @@ object EndpointsCompanion {
   private[http] def errorResponse(
       error: Error
   )(implicit lc: LoggingContextOf[InstanceUUID with RequestID]): domain.ErrorResponse = {
-    val (status, errorMsg): (StatusCode, String) = error match {
-      case InvalidUserInput(e) => StatusCodes.BadRequest -> e
-      case ParticipantServerError(grpcStatus, d) =>
-        grpcStatus.asAkkaHttpForJsonApi -> s"$grpcStatus${d.cata((": " + _), "")}"
-      case ServerError(reason) =>
-        logger.error(s"Internal server error occured", reason)
-        StatusCodes.InternalServerError -> "HTTP JSON API Server Error"
-      case Unauthorized(e) => StatusCodes.Unauthorized -> e
-      case NotFound(e) => StatusCodes.NotFound -> e
-    }
-    domain.ErrorResponse(errors = List(errorMsg), warnings = None, status = status)
+    val ((status, errorMsg), ledgerApiErrorOpt) =
```

Comment:
Instead of this tuple extraction just direcly create the ErrorResponse in the different cases?